utilities/src/main/java/io/syndesis/qe/utils/SampleDbConnectionManager.java

```diff
@@ -3,59 +3,113 @@
 import java.sql.Connection;
 import java.sql.DriverManager;
 import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
 import java.util.Properties;
 
+import io.fabric8.kubernetes.api.model.Pod;
 import io.fabric8.kubernetes.client.LocalPortForward;
+import lombok.Data;
 import lombok.extern.slf4j.Slf4j;
 
 /**
  * Nov 15, 2017 Red Hat
  *
- * @author tplevko@redhat.com
+ * @author tplevko@redhat.com, sveres@redhat.com
  */
 @Slf4j
 public class SampleDbConnectionManager {
 
-    private static LocalPortForward localPortForward = null;
-    private static Connection dbConnection = null;
+    private static Map<String, DbWrapper> connectionsInfoMap = new HashMap<>();
 
+    public static Connection getConnection(String dbType) {
 
-    private SampleDbConnectionManager() {
+        DbWrapper wrap = new DbWrapper(dbType);
```

Comment:
This is definitely better than previous version, just one last thing. You do not need to work with new wrap at all on this level.

getConnection(...) should be straight forward method like this:

1. try to get connection from static map connectionsInfoMap
2. check if it exists - yes? return it and exit. no? call method for connection creation like createConnection(dbType)
3. since you created it in previous step, now return connection from static map connectionsInfoMap

And in method createConnection(dbType) all you have to do is to create correct connection with help of switch and take advantage of static map parameter connectionsInfoMap.

The point is that you are sending instance of wrap class all over the place but in fact you do not need it at all - you only create it once and insert it into static map and then access to connector via that map.